features/layer_lock.h

```diff
@@ -82,7 +82,36 @@ void layer_lock_invert(uint8_t layer);
 //   }
 void layer_lock_set_user(layer_state_t locked_layers);
 
+#if LAYER_LOCK_IDLE_TIMEOUT > 0 // Layer Lock Timer
+
+// If you want to automatically disable layer lock if no keys are pressed after a
+// configurable timeout, define LAYER_LOCK_IDLE_TIMEOUT in your config.h file.
+// Also add the layer_lock_timer_task function under your matrix_scan_user function,
+// most likely in keymap.c
+
+// Timeout values between 100 and 32767 (in milliseconds) will use a 16 bit timer, while
+// values above 32767 will use a 32 bit timer, which uses slightly more space on
+// your MCU.
+
+    void layer_lock_timer_task(void);
+
+    void layer_lock_all_off(void);
+
+        #if LAYER_LOCK_IDLE_TIMEOUT > 100 && LAYER_LOCK_IDLE_TIMEOUT <= 32767
+
+            uint16_t layer_lock_timer_read(void);
```

Comment:
I don't think these layer lock timer functions need to be exposed to the user. Make them "private" to the library by declaring them as `static` functions in layer_lock.c and remove mention of them in layer_lock.h.

Reply:
Those separate functions aren't necessary any longer. I only added them to easily flip between 16- and 32-bit timers so I didn't have to clutter up the other functions with both timer_read() and timer_read32(). Since we're doing just 32-bit now I've only added two functions to your original feature, layer_lock_task and layer_lock_all_off.